src/lib.rs

```diff
@@ -554,9 +606,22 @@ impl Trait {
             Copy => quote! {},
             Debug => quote! { #pattern => ::core::fmt::Formatter::write_str(__f, #debug_name), },
             Eq => quote! {},
-            Hash => quote! { #pattern => (), },
+            Hash => {
+                // Add hashing the variant if this is an `enum`.
+                let discriminant = if variants.is_some() {
+                    let type_ = self.path();
+                    Some(quote! { #type_::hash(&::core::mem::discriminant(self), __state); })
```

Comment:
`type_` vs `path`, comparing to https://github.com/ModProg/derive-where/pull/15#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R541-R546

Reply:
Good catch!

Reply:
Done.